### PR TITLE
 Change the texture sample type for the cube example from Float to Uint

### PR DESCRIPTION
**Pull request by igowen**
this is a rework of #764. i think wgpu supports everything we need for this now!

### CODE REVIEW

File: examples/cube/main.rs [hunk 1]
```diff
@@ -204,22 +192,13 @@ impl framework::Example for Example {
             &texels,
             wgpu::ImageDataLayout {
                 offset: 0,
-                bytes_per_row: Some(std::num::NonZeroU32::new(4 * size).unwrap()),
-                rows_per_image: None,
+                bytes_per_row: Some(std::num::NonZeroU32::new(size).unwrap()),
+                rows_per_image: Some(std::num::NonZeroU32::new(size).unwrap()),
```
Comment (kvark) [inline examples/cube/main.rs:196]: shouldn't be needed

File: examples/cube/shader.wgsl [hunk 2]
```diff
@@ -22,13 +22,13 @@ fn vs_main(
...
 [[group(0), binding(1)]]
-var r_color: texture_2d<f32>;
-[[group(0), binding(2)]]
-var r_sampler: sampler;
+var r_color: texture_2d<u32>;
 
 [[stage(fragment)]]
 fn fs_main(in: VertexOutput) -> [[location(0)]] vec4<f32> {
-    return textureSample(r_color, r_sampler, in.tex_coord);
+    var tex: vec4<u32> = textureLoad(r_color, vec2<i32>(in.tex_coord * 256.0), 0);
```
Comment (kvark) [inline examples/cube/shader.wgsl:29]: let's do `let tex = textureLoad(...);`

File: examples/cube/shader.wgsl [hunk 3]
```diff
@@ -22,13 +22,13 @@ fn vs_main(
...
+    var v: f32 = f32(tex.x) / 255.0;
```
Comment (kvark) [inline examples/cube/shader.wgsl:30]: `let v = ...;`

**Review by kvark — COMMENTED** [review]
Got a few nits, looks great overall!

File: examples/cube/main.rs (hunk 1 above)
Comment (igowen) [reply examples/cube/main.rs]: removed

File: examples/cube/shader.wgsl (hunk 2 above)
Comment (igowen) [reply examples/cube/shader.wgsl]: done

File: examples/cube/shader.wgsl (hunk 3 above)
Comment (igowen) [reply examples/cube/shader.wgsl]: done

**Comment by igowen** [comment]
thanks for reviewing!

**Comment by kvark** [comment]
Could you squash everything please?

**Comment by igowen** [comment]
done! sorry, still not super familiar with the whole github workflow :)

**Pull request merged**